src/main/java/org/assertj/core/internal/Urls.java

```diff
@@ -138,4 +138,21 @@ public void assertHasNoParameter(AssertionInfo info, URL actual, String name, St
     }
   }
 
+  public void assertIsEqualToWithSortedQueryParameters(AssertionInfo info, URL actual, URL expected) {
+    assertNotNull(info, actual);
+    if (expected.getQuery() == null || actual.getQuery() == null)
+      if (expected.getQuery() == null && actual.getQuery() == null)
+        return;
+      else throw failures.failure(info, shouldBeEqualToWithSortedQueryParameters(actual, expected));
+    String expected_non_query = expected.toString().substring(0, expected.toString().length() - expected.getQuery().length());
+    String actual_non_query = actual.toString().substring(0, actual.toString().length() - actual.getQuery().length());
+    if (!expected_non_query.equals(actual_non_query))
+      throw failures.failure(info, shouldBeEqualToWithSortedQueryParameters(actual, expected));
+    String[] expected_query_params = expected.getQuery().split("&");
+    String[] actual_query_params = actual.getQuery().split("&");
```

Comment:
You would have a fragment if the URL has one [with query params and fragments](https://en.wikipedia.org/wiki/URL) :
```
URI = scheme:[//authority]path[?query][#fragment]
```